resources/kubernetes/portal-app/portal-domains.ts

```diff
@@ -1,89 +1,135 @@
-    .map(
-      customer =>
-        new k8s.networking.v1.Ingress(
-          `${customer.ident.current}-ingress`,
-          {
-            metadata: {
-              name: `customer-${customer.ident.current}`,
-              namespace: namespace.metadata.name,
-              annotations: {
-                'kubernetes.io/ingress.class': 'caddy',
+  customers.map(customer => {
+    const customerDomain = customer.domain
+      ? customer.domain
+      : `${customer.ident.current}.${rootDomain}`;
 
-                'pulumi.com/skipAwait': 'true',
-              },
-              labels: {
-                customer: customer.ident.current,
-                kind: 'customer-domain',
-              },
-            },
-            spec: {
-              rules: [
-                {
-                  host: customer.domain,
-                  http: {
-                    paths: [
-                      {
-                        path: '/',
-                        pathType: 'Prefix',
-                        backend: {
-                          service: {
-                            name: portalApp.service.metadata.name,
-                            port: { number: portalApp.port },
-                          },
-                        },
+    const domainKind = customer.domain ? 'customer-domain' : 'root-domain';
+    const debitorPortalDomain = customerDomain;
+    const creditorPortalDomain = `kred.${customerDomain}`;
+    const apiDomain = `api.${customerDomain}`;
+
+    if (!customer.domain) {
+      new gcp.dns.RecordSet(
+        `${customer.ident.current}-creditor-portal`,
+        {
+          managedZone: zone.name,
+          name: creditorPortalDomain,
+          type: 'A',
+          ttl: 300,
+          rrdatas: [ingressIpAddress],
+        },
+        { provider },
+      );
+      new gcp.dns.RecordSet(
+        `${customer.ident.current}-debitor-portal`,
+        {
+          managedZone: zone.name,
+          name: debitorPortalDomain,
+          type: 'A',
+          ttl: 300,
+          rrdatas: [ingressIpAddress],
+        },
+        { provider },
+      );
+
+      new gcp.dns.RecordSet(
+        `${customer.ident.current}-api`,
+        {
+          managedZone: zone.name,
+          name: apiDomain,
+          type: 'A',
+          ttl: 300,
+          rrdatas: [ingressIpAddress],
+        },
+        { provider },
+      );
+    }
+
+    new k8s.networking.v1.Ingress(
+      `${customer.ident.current}-ingress`,
+      {
+        metadata: {
+          name: `customer-${customer.ident.current}`,
+          namespace: namespace.metadata.name,
+          annotations: {
+            'kubernetes.io/ingress.class': 'caddy',
+
+            'pulumi.com/skipAwait': 'true',
+          },
+          labels: {
+            customer: customer.ident.current,
+            kind: domainKind,
+          },
+        },
+        spec: {
+          rules: [
+            {
+              host: `kred.${customerDomain}`,
```

Comment:
is `kred.{customerDomain}` the url we want to use? the pulumi var says `creditor.fpx.no`.